server/pulp/server/async/tasks.py

```diff
@@ -255,7 +255,7 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         # To avoid the race condition where __call__ method below is called before
         # this change is propagated to all db nodes, using an 'upsert' here and setting
         # the task state to 'waiting' only on an insert.
-        task_status.save(fields_to_set_on_insert=['state', 'start_time'])
+        task_status.save(fields_to_set_on_insert=['state', 'start_time'], safe=True)
```

Comment:
Great job setting up the testbed! You've proven that specifying `{'fsync': True}` allows a write request to return before hitting all the nodes, even though the replica set is configured for a safer write concern. IMHO, this is a bug in mongoDB because the replica set is configured for one level of safety, but a client can override it. That doesn't seem right. If client software needs to force a flush of all writes to disk, it's forced to "mirror" the specification of the safety level which an administrator has already configured mongoDB. That is redundant and potentially unsafe if done incorrectly (as you've proven). I remember you mentioning that they brought back the `w:majority` setting in 3.0, which keeps the client out of the business of specifying the replication level, but we still use mongoDB 2.x. Again, great investigation work. 

To work around this bug, the only thing we can do is have Pulp [compute the write concern as you suggest](https://github.com/pulp/pulp/pull/1864/files#r30699492). Let's do that.